silverback/middlewares.py

```diff
@@ -77,6 +93,21 @@ def post_execute(self, message: TaskiqMessage, result: TaskiqResult):
             f"- {result.execution_time:.3f}s ({percentage_time:.1f}%)"
         )
 
+    async def post_save(self, message: TaskiqMessage, result: TaskiqResult):
+        if not self.persistence:
+            return
+
+        handler_id, block_number, log_index = resolve_task(message)
+
+        handler_result = HandlerResult.from_taskiq(
+            self.ident, handler_id, block_number, log_index, result
+        )
+
+        try:
+            await self.persistence.add_result(handler_result)
```

Comment:
random question: does the middleware act on the client or the worker?

Reply:
Both.  pre_send and post_send are client side, and pre_execute, post_execute, and post_save are worker-side.  It never comes back to the client after processing, though.